content/main.js

```diff
@@ -458,4 +458,16 @@ var tblatex = {
       var tb = document.getElementById("composeToolbar2");
       tb.setAttribute("defaultset", tb.getAttribute("defaultset")+",tblatex-button-1");
     }, false);
+
+  window.addEventListener("unload",
+      // Remove all cached images on closing the composer window
+    function() {
+      for (var key in g_image_cache) {
+        var f = Components.classes["@mozilla.org/file/local;1"].createInstance(Components.interfaces.nsIFile);
+        try {
+          f.initWithPath(g_image_cache[key]);
```

Comment:
should the entry be removed from g_image_cache?

Reply:
Probably yes. I admit that I don't know much (actually virtually nothing) about TB internals...

How would I do that? `delete g_image_cache[key]`?

Reply:
`g_image_cache` is managed by the addon, so it's not an internal Thunderbird variable

what you wrote is the right way to delete it, and I would actually do it, since Thunderbird recycles composition windows so global state might be preserved from one composition window to another